### PR TITLE
Fix possible leak in SslHandler if wrap(...) throws.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We can end up with a buffer leak if SSLEngine.wrap(...) throws.

Modifications:

Correctly release the ByteBuf if SSLEngine.wrap(...) throws.

Result:

Fixes [#7337].

### CODE REVIEW

File: handler/src/main/java/io/netty/handler/ssl/SslHandler.java [hunk 1]
```diff
@@ -801,6 +804,10 @@ private void wrap(ChannelHandlerContext ctx, boolean inUnwrap) throws SSLExcepti
                 }
             }
         } finally {
+            // Ownership of buffer was not transfered, release it.
```
Comment (Scottmitch) [inline handler/src/main/java/io/netty/handler/ssl/SslHandler.java]: `transfered ` -> `transferred`

**Comment by eun-ice** [comment]
I have seen SSLEngine.wrap throw with OpenSSL in production and some weird half-closed SSL states.
So thank you for this fix!

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (92b786e2f33a44f16be224305fe4fb716422b14a). 4.0 is not affected

**Comment by jchambers** [comment]
I'm trying to figure out if this might have been the cause of some leaks I've been chasing down. My understanding is that:

1. This problem has been around for at least six months, and possibly longer.
2. In practical terms, this problem is most likely to manifest on a lossy connection, and generally should not me an issue on high-quality, low-loss connections.

Is that accurate? If it is, then I think this solves our problem. Thanks kindly!

**Comment by normanmaurer** [comment]
Yes I think it’s correct

**Comment by jchambers** [comment]
Thank you!

**Pull request closed without merging**